File: src/main/java/org/example/bankramenserver/domain/auth/service/StateService.java
package org.example.bankramenserver.domain.auth.service;

import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.stereotype.Service;

import java.util.UUID;
import java.util.concurrent.TimeUnit;

@Service
@RequiredArgsConstructor
public class StateService {

    private final RedisTemplate<String, String> redisTemplate;

    @Value("${oauth.state.expiration}")
    private long stateExpiration;

    @Value("${oauth.state.prefix}")
    private String PREFIX;

    public String generateState() {
        String state = UUID.randomUUID().toString();
        redisTemplate.opsForValue()
                .set(PREFIX + state, state, stateExpiration, TimeUnit.SECONDS);
        return state;
    }

    public boolean validateState(String state) {
        String key = PREFIX + state;

        String saved = redisTemplate.opsForValue().getAndDelete(key);

        return saved != null;
    }
}

File: src/main/kotlin/org/example/bankramenserver/domain/auth/service/StateService.kt
package org.example.bankramenserver.domain.auth.service
import org.springframework.beans.factory.annotation.Value
import org.springframework.data.redis.core.RedisTemplate
import org.springframework.stereotype.Service
import java.util.UUID
import java.util.concurrent.TimeUnit
@Service class StateService(private val redisTemplate: RedisTemplate<String, String>, @Value("\${oauth.state.expiration}") private val stateExpiration: Long, @Value("\${oauth.state.prefix}") private val prefix: String) { fun generateState() = UUID.randomUUID().toString().also { redisTemplate.opsForValue().set(prefix + it, it, stateExpiration, TimeUnit.SECONDS) }; fun validateState(state: String) = redisTemplate.opsForValue().getAndDelete(prefix + state) != null }
